ts/packages/ts/src/commands/UpsertCommand.ts

```diff
@@ -60,6 +60,10 @@ export class UpsertCommand extends Command<void> {
           );
         }
 
+        if (cfgu.lazy) {
+          throw new ConfigError('invalid config key', `key "${key}" is lazy and cannot be set directly`, errorScope);
```

Comment:
How about this to keep it similar to the template error:
```suggestion
          throw new ConfigError('invalid config key', `keys declared as lazy mustn't have a value`, errorScope);
```